packages/manager/src/features/CloudPulse/Alerts/NotificationChannels/CreateChannel/NotificationChannelTypeSelect.tsx

```diff
@@ -29,19 +33,22 @@ export interface NotificationChannelTypeSelectProps {
 
 export const NotificationChannelTypeSelect = React.memo(
   (props: NotificationChannelTypeSelectProps) => {
-    const { error, handleChannelTypeChange, value, options, onBlur } = props;
+    const { disabled, error, handleChannelTypeChange, value, options, onBlur } =
+      props;
 
     return (
       <Autocomplete
         data-testid="channel-type-select"
+        disableClearable={disabled}
+        disabled={disabled}
         errorText={error}
         label="Type"
         onBlur={onBlur}
         onChange={(_, selected: Item<string, ChannelType>, reason) => {
-          if (selected) {
+          if (selected && handleChannelTypeChange) {
             handleChannelTypeChange(selected.value);
```

Comment:
```suggestion
            handleChannelTypeChange?.(selected.value);
```

instead of checking above...